src/mqtt_pal.c

```diff
@@ -87,7 +87,7 @@ ssize_t mqtt_pal_sendall(mqtt_pal_socket_handle fd, const void* buf, size_t len,
 }
 
 ssize_t mqtt_pal_recvall(mqtt_pal_socket_handle fd, void* buf, size_t bufsz, int flags) {
-    const void const *start = buf;
+    const void *start = buf;
```

Comment:
Here `start` is a constant pointer to constant data (since `buf` shouldn't be modified). As you pointed out, my original declaration was incorrect, but I'd still like to keep the double-constness.

I think this is the proper declaration:
```c
const void *const start = buf;
```